### ISSUE TITLE
Copy-paste doesn't work on Wayland

### ISSUE THREAD

**Issue by PureTryOut** [Lea-fish/Leafish#110]
On X11 however it works fine. This makes it hard to enter my password and I have to switch to an X11 session :cry: 

Tested on KDE Plasma 5.22

**Comment by kepet19** [comment]
Hey i was looking around found out: 
Alacritty has a fork off clipboard called copypasta that supports wayland. 
https://github.com/alacritty/copypasta 
Then we don't need to change anything than the dep maybe?

**Comment by PureTryOut** [comment]
Good find. `rust-clipboard` seems unmaintained as well, I'll give the fork a shot.

**Comment by kepet19** [comment]
I tried it the api has change a little bit, but it works perfekt on  Sway.

**Comment by PureTryOut** [comment]
Tried it as well, and so far no luck :thinking:

**Comment by kepet19** [comment]
Ohh Yeah I forgot about it, I also had problems copy paste with alacritty.
https://www.reddit.com/r/swaywm/comments/b0r9rt/is_there_any_solution_to_copypaste_in_alacritty/
This should not be the work around I think, I forgot I installed it.

**Comment by PureTryOut** [comment]
Yeah it definitely requires more work than just adapting the API (although updating the dep is still a good thing, I'll make a PR for that anyhow).

https://github.com/alacritty/copypasta/issues/27
https://github.com/alacritty/copypasta/blob/master/src/wayland_clipboard.rs#L36

We need to give the clipboard a pointer to the Wayland display.

**Comment by kepet19** [comment]
```
    #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
    pub unsafe fn new(display: Option<*mut c_void>) -> Self {
        match display {
            Some(display) => {
                let (selection, clipboard) =
                    wayland_clipboard::create_clipboards_from_external(display);
                Self { clipboard: Box::new(clipboard), selection: Some(Box::new(selection)) }
            },
            None => Self::default(),
        }
    }
```
https://github.com/alacritty/alacritty/blob/531e494cf9a90de91dbbb84cb6cfc3158b78f1f2/alacritty/src/clipboard.rs


Not sure if we can do it this way through winit
https://github.com/rust-windowing/winit/issues/162#issuecomment-416025611

**Comment by PureTryOut** [comment]
Fixed in 0dcb82e
